cmd/user-list/main_test.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/edulinq/autograder/internal/api/users"
+	"github.com/edulinq/autograder/internal/cmd"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+// Use the common main for all tests in this package.
+func TestMain(suite *testing.M) {
+	cmd.CMDServerTestingMain(suite)
+}
+
+func TestServerUserListBase(test *testing.T) {
+	testCases := []struct {
+		expectedNumUsers int
+		expectedExitCode int
+	}{
+		{10, 0},
+	}
+
+	serverUsersMap := db.MustGetServerUsers()
+
+	expectedEmailList := make([]string, 0, len(serverUsersMap))
+	for _, user := range serverUsersMap {
+		expectedEmailList = append(expectedEmailList, user.Email)
+	}
+
+	for i, testCase := range testCases {
+		args := []string{}
+
+		stdout, stderr, exitCode, err := cmd.RunCMDTest(test, main, args)
+		if err != nil {
+			test.Errorf("Case %d: CMD run returned an error: '%v'.", i, err)
+			continue
+		}
+
+		if len(stderr) > 0 {
+			test.Errorf("Case %d: CMD has content in stderr: '%s'.", i, stderr)
+			continue
+		}
+
+		if testCase.expectedExitCode != exitCode {
+			test.Errorf("Case %d: Unexpected exit code. Expected: '%d', Actual: '%d'.", i, testCase.expectedExitCode, exitCode)
+			continue
+		}
+
+		var responseContent users.ListResponse
+		util.MustJSONFromString(stdout, &responseContent)
+
+		actualNumUsers := len(responseContent.Users)
+		if testCase.expectedNumUsers != actualNumUsers {
+			test.Errorf("Case %d: Unexpected number of server users. Expected: '%d', Actual: '%d'.", i, testCase.expectedNumUsers, actualNumUsers)
+			continue
+		}
+
+		actualEmailList := make([]string, 0, len(responseContent.Users))
+		for _, user := range responseContent.Users {
+			actualEmailList = append(actualEmailList, user.Email)
+		}
```

Comment:
This is not sufficient.

For CLI tests, one of the major things you need to test is that the formatting is correct.
THis is important, because CLI programs are used in scripts, and slight changes in formatting there can break scripts.

I recommend you first checking the JSON output by a direct map comparison (probably via reflect.DeepEqual()),
then checking table output via string comparison.

Reply:
db.MustGetServerUsers gives back model.ServerUser while the user-list cmd gives back core.ServerUserInfo.
I suppose I could directly call the api for the expected output, but since that's what the cmd is doing that feels a bit pointless. 
Or I could manually build up the map of serverusers after the cmd calls the api by calling GetServerUser on each email. 

What do you think?